File: _includes/app/basic-sample.clj
;; Import the driver.
(use 'clojure.java.jdbc)

;; Connect to the "bank" database..
(def db-spec {:subprotocol "postgresql"
              :subname "//localhost:26257/bank"
              :user "maxroach"
              :password ""})

(with-db-connection [conn db-spec]

  ;; Insert two rows into the "accounts" table.
  (insert! conn :accounts {:id 1 :balance 1000} {:id 2 :balance 250})

  ;; Print out the balances.
  (println "Initial balances:")
  (->> (query conn ["SELECT id, balance FROM accounts"])
       (map println)
       doall)

;; The database connection is automatically closed by with-db-connection.
)

File: _includes/app/basic-sample.rs
extern crate postgres;

use postgres::{Connection, TlsMode};

fn main() {
    let conn = Connection::connect("postgresql://maxroach@localhost:26257/bank", TlsMode::None)
        .unwrap();

    // Insert two rows into the "accounts" table.
    conn.execute(
        "INSERT INTO accounts (id, balance) VALUES (1, 1000), (2, 250)",
        &[],
    ).unwrap();

    // Print out the balances.
    println!("Initial balances:");
    for row in &conn.query("SELECT id, balance FROM accounts", &[]).unwrap() {
        let id: i64 = row.get(0);
        let balance: i64 = row.get(1);
        println!("{} {}", id, balance);
    }
}
